server/services/store/sqlstore/workspaces.go

```diff
@@ -150,30 +156,26 @@ func (s *SQLStore) GetWorkspaceCount() (int64, error) {
 func (s *SQLStore) GetUserWorkspaces(userID string) ([]model.UserWorkspace, error) {
 	var query sq.SelectBuilder
 
-	if s.dbType == mysqlDBType {
-		query = s.getQueryBuilder().
-			Select("Channels.ID", "Channels.DisplayName", "COUNT(focalboard_blocks.id)").
-			From("focalboard_blocks").
-			Join("ChannelMembers ON focalboard_blocks.workspace_id = ChannelMembers.ChannelId").
-			Join("Channels ON ChannelMembers.ChannelId = Channels.Id").
-			Where(sq.Eq{"ChannelMembers.UserId": userID}).
-			Where(sq.Eq{"focalboard_blocks.type": "board"}).
-			Where(sq.Like{"focalboard_blocks.fields": "%\"isTemplate\":false%"}).
-			GroupBy("Channels.Id", "Channels.DisplayName")
-	} else {
-		query = s.getQueryBuilder().
-			Select("channels.id", "channels.displayname", "count(focalboard_blocks.id)").
-			From("focalboard_blocks").
-			Join("channelmembers ON focalboard_blocks.workspace_id = channelmembers.channelid").
-			Join("channels ON channelmembers.channelid = channels.id").
-			Where(sq.Eq{"channelmembers.userid": userID}).
-			Where(sq.Eq{"focalboard_blocks.type": "board"}).
-			Where("focalboard_blocks.fields ->> 'isTemplate' = 'false'").
-			GroupBy("channels.id", "channels.displayname")
+	query = s.getQueryBuilder().
+		Select("Channels.ID", "Channels.DisplayName", "COUNT(focalboard_blocks.id)").
+		From("focalboard_blocks").
+		Join("ChannelMembers ON focalboard_blocks.workspace_id = ChannelMembers.ChannelId").
+		Join("Channels ON ChannelMembers.ChannelId = Channels.Id").
+		Where(sq.Eq{"ChannelMembers.UserId": userID}).
+		Where(sq.Eq{"focalboard_blocks.type": "board"})
+
+	switch s.dbType {
+	case mysqlDBType:
+		query = query.Where(sq.Like{"focalboard_blocks.fields": "%\"isTemplate\":false%"})
+	case postgresDBType:
+		query = query.Where("focalboard_blocks.fields ->> 'isTemplate' = 'false'")
+	default:
+		return nil, fmt.Errorf("GetUserWorkspaces - %w", errUnsupportedDatabaseError)
 	}
 
-	rows, err := query.Query()
+	query = query.GroupBy("Channels.Id", "Channels.DisplayName")
```

Comment:
nit: the `GroupBy` can be added to the query in the initial query builder. I don't think is needed to do it later.

Reply:
Interesting! I didn't know the clauses were order independent! Nice!